nni/algorithms/compression/v2/pytorch/pruning/tools/sparsity_allocator.py

```diff
@@ -36,6 +36,8 @@ def generate_sparsity(self, metrics: Dict[str, Tensor]) -> Dict[str, Dict[str, T
                 threshold = torch.topk(metric.view(-1), prune_num, largest=False)[0].max()
             mask = torch.gt(metric, threshold).type_as(metric)
             masks[name] = self._expand_mask(name, mask)
+            if self.continuous_mask:
```

Comment:
what's this for?

Reply:
e.g., if we set fine-grained pre-masks to a channel-wise pruner, then the generated masks will be channel-wise masks and may not inherit the fine-grained pre-masks. This line is for inheriting the pre-maks.

Reply:
Looks like it is not related to the `import path`? If so, suggest to split into two PRs, so that the comments can be much clear when merged.

Reply:
split to pr #4428